test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperOutputTest.cs

```diff
@@ -209,6 +301,56 @@ public void GenerateEndTag_ReturnsNothingIfSelfClosing()
             Assert.Empty(output);
         }
 
+        [Fact]
+        public void SupressOutput_Sets_TagName_Content_PreContent_PostContent_ToNull()
+        {
+            // Arrange
+            var tagHelperOutput = new TagHelperOutput("p")
+            {
+                PreContent = "Pre Content",
+                Content = "Content",
+                PostContent = "Post Content"
+            };
+
+            // Act
+            tagHelperOutput.SupressOutput();
+
+            // Assert
+            Assert.Null(tagHelperOutput.TagName);
+            Assert.Null(tagHelperOutput.PreContent);
+            Assert.Null(tagHelperOutput.Content);
+            Assert.Null(tagHelperOutput.PostContent);
+        }
+
+        [Fact]
+        public void SupressOutput_PreventsTagOutput()
+        {
+            // Arrange
+            var tagHelperOutput = new TagHelperOutput("p",
+                attributes: new Dictionary<string, string>
+                {
+                    { "class", "btn" },
+                    { "something", "   spaced    " }
+                })
+            {
+                PreContent = "Pre Content",
+                Content = "Content",
+                PostContent = "Post Content"
+            };
+
+            // Act
+            tagHelperOutput.SupressOutput();
+
+            var output = tagHelperOutput.GenerateStartTag() +
+                         tagHelperOutput.GeneratePreContent() +
+                         tagHelperOutput.GenerateContent() +
+                         tagHelperOutput.GeneratePostContent() +
+                         tagHelperOutput.GenerateEndTag();
+
+            // Assert
+            Assert.Empty(output);
```

Comment:
nit: failures will be hard to debug.  split into multiple assertions e.g. `Assert.Empty(tagHelperOutput.GenerateStartTag();`